src/main/java/org/testng/ITestRunnerFactory2.java

```diff
@@ -0,0 +1,28 @@
+package org.testng;
+
+import org.testng.xml.XmlTest;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A factory for TestRunners to be used by SuiteRunners.
+ */
+public interface ITestRunnerFactory2 extends ITestRunnerFactory {
```

Comment:
It would work because in the long run, we will be doing the following (once we are on JDK8)

1. Deprecate the below method in `org.testng.ITestRunnerFactory`

```java
TestRunner newTestRunner(ISuite suite, XmlTest test, Collection<IInvokedMethodListener> listeners, List<IClassListener> classListeners);
```

2. Merge the method 

```java
TestRunner newTestRunner(ISuite suite, XmlTest test,
                             Collection<IInvokedMethodListener> listeners,
                             List<IClassListener> classListeners,
                             Map<Class<? extends IDataProviderListener>, IDataProviderListener> dataProviderListeners);
```
from `org.testng.ITestRunnerFactory2` into `org.testng.ITestRunnerFactory` and provide a default implementation (which if users want, can override)

3. Refactor TestNG to invoke the new variant of `newTestRunner` and not the deprecated one.